src/Neo/SmartContract/Native/NeoToken.cs

```diff
@@ -272,18 +272,35 @@ internal override async ContractTask PostPersistAsync(ApplicationEngine engine)
 
             if (ShouldRefreshCommittee(engine.PersistingBlock.Index, m))
             {
-                BigInteger voterRewardOfEachCommittee = gasPerBlock * VoterRewardRatio * VoteFactor * m / (m + n) / 100; // Zoom in VoteFactor times, and the final calculation should be divided VoteFactor
+                var withFlatRewards = engine.IsHardforkEnabled(Hardfork.HF_Faun);
+                BigInteger voterRewardOfEachCommittee =
+                    withFlatRewards ?
+                    gasPerBlock * VoterRewardRatio * VoteFactor / 100 :                 // Total reward for voters when it's flat rewards
+                    gasPerBlock * VoterRewardRatio * VoteFactor * m / (m + n) / 100;    // Zoom in VoteFactor times, and the final calculation should be divided VoteFactor
+                var votersCount = (BigInteger)engine.SnapshotCache[_votersCount];
+
                 for (index = 0; index < committee.Count; index++)
                 {
                     var (publicKey, votes) = committee[index];
-                    var factor = index < n ? 2 : 1; // The `voter` rewards of validator will double than other committee's
-                    if (votes > 0)
+
+                    if (withFlatRewards)
                     {
```

Comment:
We need skip votersCount = 0, otherwise you even can't launch a privatenet when `HF_Faun` is enabled.
```suggestion
                    if (withFlatRewards)
                    {
                        if (votersCount.IsZero)
                        {
                            // Skip reward distribution when there are no voters
                            continue;
                        }
```